exporter/datadogexporter/internal/model/translator/exponential_histograms_translator_test.go

```diff
@@ -0,0 +1,184 @@
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package translator // import "github.com/open-telemetry/opentelemetry-collector-contrib/exporter/datadogexporter/internal/model/translator"
+
+import (
+	"context"
+	"math"
+	"testing"
+	"time"
+
+	"github.com/DataDog/datadog-agent/pkg/quantile/summary"
+	"github.com/stretchr/testify/assert"
+	"go.opentelemetry.io/collector/model/pdata"
+	"go.uber.org/zap"
+)
+
+const (
+	acceptableFloatError = 1e-12
+)
+
+func TestExponentialHistogramToDDSketch(t *testing.T) {
+	ts := pdata.NewTimestampFromTime(time.Now())
+	point := pdata.NewExponentialHistogramDataPoint()
+	point.SetScale(6)
+
+	point.SetCount(30)
+	point.SetZeroCount(10)
+	point.SetSum(math.Pi)
+
+	point.Negative().SetOffset(2)
+	point.Negative().SetBucketCounts([]uint64{3, 2, 5})
+
+	point.Positive().SetOffset(3)
+	point.Positive().SetBucketCounts([]uint64{1, 1, 1, 2, 2, 3})
+
+	point.SetTimestamp(ts)
+
+	tr := newTranslator(t, zap.NewNop())
+
+	sketch, err := tr.exponentialHistogramToDDSketch(point, true)
```

Comment:
In my (brief) experience working on this, I think it's more maintainable if we call `MapMetrics` on tests instead of the unexported type-specific functions; we won't have to change the tests each time we update the implementation

Reply:
I agree for the example below, but not for this one, as this test specifically aims to check that the ExpHistogram -> DDSketch conversion is lossless (ie. all buckets are mapped without error in the DDSketch). If I use `MapMetrics` here, I'll get the final Agent sketch as the result, which won't let me test that.

Reply:
You are right 🤦 Sorry! Ignore this one